core/block.go

```diff
@@ -14,23 +19,134 @@ type Block struct {
 	// The StarkNet address of the sequencer who created this block
 	SequencerAddress *felt.Felt
 	// The time the sequencer created this block before executing transactions
-	Timestamp uint64
+	Timestamp *felt.Felt
 	// The number of transactions in a block
-	TransactionCount uint64
+	TransactionCount *felt.Felt
 	// A commitment to the transactions included in the block
 	TransactionCommitment *felt.Felt
 	// The number of events
-	EventCount uint64
+	EventCount *felt.Felt
 	// A commitment to the events produced in this block
 	EventCommitment *felt.Felt
 	// The version of the StarkNet protocol used when creating this block
-	ProtocolVersion uint64
+	ProtocolVersion *felt.Felt
 	// Extraneous data that might be useful for running transactions
 	ExtraData *felt.Felt
 }
 
-func (b *Block) Hash() *felt.Felt {
-	// Todo: implement pedersen hash as defined here
-	// https://docs.starknet.io/documentation/develop/Blocks/header/#block_hash
-	return nil
+type blockHashMetaInfo struct {
+	First07Block             uint64     // First block that uses the post-0.7.0 block hash algorithm
+	UnverifiableRange        []uint64   // Range of blocks that are not verifiable
+	FallBackSequencerAddress *felt.Felt // The sequencer address to use for blocks that do not have one
+}
+
+type UnverifiableBlockError struct {
+	input string
```

Comment:
Let's make the field name a bit more descriptive like `blockNumber`. I think we can also use a `uint64` here, allowing us to skip the string conversion on line 93.